lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexer.java

```diff
@@ -14,38 +18,37 @@
 import com.kmwllc.lucille.core.Indexer;
 import com.kmwllc.lucille.message.IndexerMessenger;
 import com.typesafe.config.Config;
-import io.grpc.ConnectivityState;
-import io.pinecone.PineconeClient;
-import io.pinecone.PineconeClientConfig;
-import io.pinecone.PineconeConnection;
-import io.pinecone.proto.UpdateRequest;
-import io.pinecone.proto.UpsertRequest;
-import io.pinecone.proto.Vector;
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import static io.pinecone.commons.IndexInterface.buildUpsertVectorWithUnsignedIndices;
 
 public class PineconeIndexer extends Indexer {
 
   private static final Logger log = LoggerFactory.getLogger(PineconeIndexer.class);
 
-  private final PineconeClient client;
-  private final String index;
+  private final Pinecone client;
+  private final Index index;
+  private final String indexName;
   private final Map<String, Object> namespaces;
   private final Set<String> metadataFields;
   private final String mode;
-  private PineconeConnection connection;
   private final String defaultEmbeddingField;
 
   public PineconeIndexer(Config config, IndexerMessenger messenger, String metricsPrefix) {
     super(config, messenger, metricsPrefix);
-    this.index = config.getString("pinecone.index");
+    this.indexName = config.getString("pinecone.index");
     this.namespaces = config.hasPath("pinecone.namespaces") ? config.getConfig("pinecone.namespaces").root().unwrapped() : null;
     this.metadataFields = new HashSet<>(config.getStringList("pinecone.metadataFields"));
     this.mode = config.hasPath("pinecone.mode") ? config.getString("pinecone.mode") : "upsert";
     this.defaultEmbeddingField = ConfigUtils.getOrDefault(config, "pinecone.defaultEmbeddingField", null);
-    PineconeClientConfig configuration = new PineconeClientConfig().withApiKey(config.getString("pinecone.apiKey"))
-        .withEnvironment(config.getString("pinecone.environment")).withProjectName(config.getString("pinecone.projectName"))
-        .withServerSideTimeoutSec(config.getInt("pinecone.timeout"));
-    this.client = new PineconeClient(configuration);
-
+    // providing environment is only for pod-based indexes, but pinecone is moving to serverless; cheaper and more performant
+    // moving from pod based to serverless https://docs.pinecone.io/guides/indexes/migrate-a-pod-based-index-to-serverless
+    // api key is now project specific, so no need for project more info here https://docs.pinecone.io/guides/get-started/key-concepts
+//    PineconeConfig configuration = new PineconeConfig(config.getString("pinecone.apiKey"))
+//        .withEnvironment(config.getString("pinecone.environment")).withProjectName(config.getString("pinecone.projectName"))
+//        .withServerSideTimeoutSec(config.getInt("pinecone.timeout"));
```

Comment:
Awesome, thanks for including this information for me, super good to know. I think you can remove these comments before merging though, that info doesn't need to stay in the codebase. Maybe copy/paste them as a comment on the PR?